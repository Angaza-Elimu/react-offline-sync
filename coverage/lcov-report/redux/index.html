
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for redux</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> redux</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.79% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>190/214</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">80.15% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>105/131</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">78.57% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>33/42</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">88.4% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>160/181</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
    </div>
    <div class='status-line high'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="actionCreators.ts"><a href="actionCreators.ts.html">actionCreators.ts</a></td>
	<td data-value="93.33" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 93%"></div><div class="cover-empty" style="width: 7%"></div></div>
	</td>
	<td data-value="93.33" class="pct high">93.33%</td>
	<td data-value="15" class="abs high">14/15</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="4" class="abs medium">3/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="90" class="pct high">90%</td>
	<td data-value="10" class="abs high">9/10</td>
	</tr>

<tr>
	<td class="file high" data-value="actionTypes.ts"><a href="actionTypes.ts.html">actionTypes.ts</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="5" class="abs high">5/5</td>
	</tr>

<tr>
	<td class="file high" data-value="createNetworkMiddleware.ts"><a href="createNetworkMiddleware.ts.html">createNetworkMiddleware.ts</a></td>
	<td data-value="97.14" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 97%"></div><div class="cover-empty" style="width: 3%"></div></div>
	</td>
	<td data-value="97.14" class="pct high">97.14%</td>
	<td data-value="70" class="abs high">68/70</td>
	<td data-value="91.8" class="pct high">91.8%</td>
	<td data-value="61" class="abs high">56/61</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="16" class="abs high">16/16</td>
	<td data-value="96.72" class="pct high">96.72%</td>
	<td data-value="61" class="abs high">59/61</td>
	</tr>

<tr>
	<td class="file medium" data-value="createReducer.ts"><a href="createReducer.ts.html">createReducer.ts</a></td>
	<td data-value="76.19" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 76%"></div><div class="cover-empty" style="width: 24%"></div></div>
	</td>
	<td data-value="76.19" class="pct medium">76.19%</td>
	<td data-value="42" class="abs medium">32/42</td>
	<td data-value="54.17" class="pct medium">54.17%</td>
	<td data-value="24" class="abs medium">13/24</td>
	<td data-value="75" class="pct medium">75%</td>
	<td data-value="8" class="abs medium">6/8</td>
	<td data-value="71.43" class="pct medium">71.43%</td>
	<td data-value="35" class="abs medium">25/35</td>
	</tr>

<tr>
	<td class="file high" data-value="sagas.ts"><a href="sagas.ts.html">sagas.ts</a></td>
	<td data-value="86.59" class="pic high">
	<div class="chart"><div class="cover-fill" style="width: 86%"></div><div class="cover-empty" style="width: 14%"></div></div>
	</td>
	<td data-value="86.59" class="pct high">86.59%</td>
	<td data-value="82" class="abs high">71/82</td>
	<td data-value="78.57" class="pct medium">78.57%</td>
	<td data-value="42" class="abs medium">33/42</td>
	<td data-value="46.15" class="pct low">46.15%</td>
	<td data-value="13" class="abs low">6/13</td>
	<td data-value="88.57" class="pct high">88.57%</td>
	<td data-value="70" class="abs high">62/70</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank">istanbul</a>
                at Tue Jun 22 2021 20:09:59 GMT+0300 (East Africa Time)
            </div>
        </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    