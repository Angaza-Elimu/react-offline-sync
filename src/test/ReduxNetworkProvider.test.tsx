import React from 'react';
import { View, Text } from 'react-native';
import {
  shallow as rnShallow,
  render as rnRender,
} from 'react-native-testing-library';
import { shallow } from 'enzyme';
import {
  ReduxNetworkProvider
} from '../components/ReduxNetworkProvider';
import { connectionChange } from '../redux/actionCreators';

const dispatch = jest.fn();
const props = {
  dispatch,
  isConnected: false,
};

const getProps = (overrides = {}) => ({ ...props, ...overrides });

describe('ReduxNetworkProvider', () => {
  afterEach(() => {
    dispatch.mockClear();
  });

  describe('render', () => {
    it('has the correct structure', () => {
      const { output } = rnShallow(
        <ReduxNetworkProvider {...props}>
          <View />
        </ReduxNetworkProvider>,
      );
      expect(output).toMatchSnapshot();
    });

    it('renders the children correctly', () => {
      const { getByText } = rnRender(
        <ReduxNetworkProvider {...props}>
          <Text>Baz</Text>
        </ReduxNetworkProvider>,
      );

      const viewChild = getByText('Baz');
      expect(viewChild).toBeDefined();
    });
  });

  describe('handleConnectivityChange', () => {
    it(`dispatches a CONNECTION_CHANGE action with the new connection`, () => {
      const wrapper = shallow<ReduxNetworkProvider>(
        <ReduxNetworkProvider {...props}>
          <View />
        </ReduxNetworkProvider>,
      );
      wrapper.instance().handleConnectivityChange(true);
      expect(props.dispatch).toHaveBeenCalledWith(connectionChange(true));
      expect(props.dispatch).toHaveBeenCalledTimes(1);
    });

    it(`does NOT dispatch a CONNECTION_CHANGE action if the connection
    did not change`, () => {
      const wrapper = shallow<ReduxNetworkProvider>(
        <ReduxNetworkProvider {...getProps({ isConnected: true })}>
          <View />
        </ReduxNetworkProvider>,
      );
      wrapper.instance().handleConnectivityChange(true);
      expect(props.dispatch).not.toHaveBeenCalled();
    });
  });
});


