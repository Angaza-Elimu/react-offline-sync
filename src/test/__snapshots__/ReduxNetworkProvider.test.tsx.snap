// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReduxNetworkProvider render has the correct structure 1`] = `
<NetworkConnectivity
  customHeaders={Object {}}
  dispatch={[MockFunction]}
  httpMethod="HEAD"
  isConnected={false}
  onConnectivityChange={[Function]}
  pingInBackground={false}
  pingInterval={0}
  pingOnlyIfOffline={false}
  pingServerUrl="https://www.google.com/"
  pingTimeout={10000}
  shouldPing={true}
>
  [Function]
</NetworkConnectivity>
`;
